cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -43,6 +50,7 @@ type SimulateTransactionResponse struct {
 	Results         []SimulateHostFunctionResult `json:"results,omitempty"`         // an array of the individual host function call results
 	Cost            SimulateTransactionCost      `json:"cost,omitempty"`            // the effective cpu and memory cost of the invoked transaction execution.
 	RestorePreamble *RestorePreamble             `json:"restorePreamble,omitempty"` // If present, it indicates that a prior RestoreFootprint is required
+	StateDiff       []LedgerEntryDiff            `json:"stateDiff,omitempty"`       // If present, it indicates how the state (ledger entries) will change as a result of the transaction execution.
```

Comment:
The comment indicates that it may or may not be present. When would the state diff be omitted? If a transactions results in no affect to state then I think the state diff should be an empty array rather than missing from the response. It tells a clear story that state was unaffected rather than being silent on the matter.

Reply:
The diff will be omitted when there are no changes or ... for operations other than InvokeHostFunction (i.e. Restore/Extend footprint).

> I think the state diff should be an empty array rather than missing from the response

What you suggest makes sense but we already omit pretty much every other field when empty. So, I am reluctant to do that for consistency.

Reply:
Thanks for the context, I didn't understand that the other fields were also being omitted when empty.